packages/react-component/src/main.tsx

```diff
@@ -0,0 +1,32 @@
+import { useState, useEffect, useRef } from 'react'
+import { embed } from "@genome-spy/core";
+import type { ViewSpecBase, RootSpec } from "@genome-spy/core"
+import { EmbedResult } from '@genome-spy/core/types/embedApi.js';
+
+interface IGenomeSpyProps {
+    spec: ViewSpecBase;
+    onEmbed: (api: Promise<EmbedResult>) => void; 
+}
+
+export default function GenomeSpy ({ spec, onEmbed }: IGenomeSpyProps) {
+    const embedRef = useRef<HTMLDivElement>(null)
+    const [error, setError] = useState<string | undefined>()
+
+    useEffect(() => {
+        async function embedToDoc(container: HTMLDivElement | null, config: RootSpec) {
+            try {
+               const api = await embed(container, config, { bare: true })
```

Comment:
If the component is removed from the dom, embed api's `finalize` method should be called to ensure that all resources are released, possible listeners in the document root are removed, etc.

Reply:
Thanks, should be in 093b914